components/backup/src/endpoint.rs

```diff
@@ -127,35 +163,45 @@ pub struct Endpoint<E: Engine, R: RegionInfoProvider> {
     pub(crate) region_info: R,
 }
 
-impl<E: Engine, R: RegionInfoProvider> Endpoint<E, R> {
-    pub fn new(store_id: u64, engine: E, region_info: R, db: Arc<DB>) -> Endpoint<E, R> {
-        let workers = ThreadPoolBuilder::new()
-            .name_prefix("backworker")
-            .pool_size(8) // TODO: make it configure.
-            .build();
-        Endpoint {
+/// The progress of a backup task
+pub struct Progress<R: RegionInfoProvider> {
+    store_id: u64,
+    next_start: Option<Key>,
+    end_key: Option<Key>,
+    region_info: R,
+    stopped: bool,
+}
+
+impl<R: RegionInfoProvider> Progress<R> {
+    fn new(store_id: u64, next_start: Option<Key>, end_key: Option<Key>, region_info: R) -> Self {
+        Progress {
             store_id,
-            engine,
+            next_start,
+            end_key,
             region_info,
-            workers,
-            db,
+            stopped: Default::default(),
         }
     }
 
-    fn seek_backup_range(
-        &self,
-        start_key: Option<Key>,
-        end_key: Option<Key>,
-        cancel: Arc<AtomicBool>,
-    ) -> mpsc::Receiver<BackupRange> {
+    /// forword the progress by n BackupRanges
+    ///
+    /// The number of the returned BackupRanges should <= n
+    fn forword(&mut self, n: usize) -> Vec<BackupRange> {
```

Comment:
Better to change `n` to a more understandable name

Reply:
How about `ranges`